pkg/csi/service/vanilla/controller.go

```diff
@@ -787,6 +788,10 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 	// Block Volume
 	node, err := c.nodeMgr.GetNodeByName(ctx, req.NodeId)
 	if err != nil {
+		// ControllerUnpublishVolume should succeed when the corresponding vm is already deleted.
+		if err == cnsnode.ErrNodeNotFound {
```

Comment:
We are returning `ErrNodeNotFound` only when we are unable to load node name to UUID mapping in the GetNodeByName() function. 

I think this will not help, when someone power down node VM, and delete it from the VM inventory without deleting node resource from the kubernetes cluster.

and if someone just deletes Kubernetes Node object without bringing down node VMs, we will end up in the situation, that we will not be able to create a new pod using the same volume on another node, as the volume is physically attached to some other VM on the vCenter.

Can you update PR with testing detail for the above cases?

Reply:
>I think this will not help, when someone power down node VM, and delete it from the VM inventory without deleting node resource from the kubernetes cluster.

Shouldn't Kubernetes delete the node object in this case after talking to CPI? Lets test and see whether this is handled correctly.

> and if someone just deletes Kubernetes Node object without bringing down node VMs
What is the use case for this? You mean the k8s admin accidentally deletes the wrong node in Kubernetes?

Reply:
> Shouldn't Kubernetes delete the node object in this case after talking to CPI? Lets test and see whether this is handled correctly.

Kubernetes should delete the node object, after CCM returns `false with nil error` from `InstanceExistsByProviderID`
https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/cloudprovider/vsphere/instances.go#L157-L181

and once node object is deleted we have informer to delete mapping from local cache map. need to check what interval `InstanceExistsByProviderID` is being called by Kubernetes.


> What is the use case for this? You mean the k8s admin accidentally deletes the wrong node in Kubernetes?

Yes accidental deletion of k8s node object by admin.

Reply:
> that we will not be able to create a new pod using the same volume on another node, as the volume is physically attached to some other VM on the vCenter.

Can confirm that a new pod can be created successfully using the same volume on another node. Added findings below:
https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/561#issuecomment-769993798